lib/note-content-editor.tsx

```diff
@@ -167,6 +173,86 @@ class NoteContentEditor extends Component<Props> {
     }
   };
 
+  completionProvider: (
+    selectedNoteId: T.EntityId | null
+  ) => languages.CompletionItemProvider = (selectedNoteId) => {
+    return {
+      triggerCharacters: ['['],
+
+      provideCompletionItems(model, position, context, token) {
+        const line = model.getLineContent(position.lineNumber);
+        if (!line.includes('[')) {
+          return null;
+        }
+
+        const precedingOpener = line.lastIndexOf('[', position.column);
+        const precedingCloser = line.lastIndexOf(']', position.column);
+        const precedingBracket =
+          precedingOpener >= 0 && precedingCloser < precedingOpener
+            ? precedingOpener
+            : -1;
+
+        const soFar =
+          precedingBracket >= 0
+            ? line.slice(precedingBracket + 1, position.column)
+            : '';
+
+        const notes = searchNotes(
+          {
+            searchTerms: getTerms(soFar),
+            excludeIDs: selectedNoteId ? [selectedNoteId] : [],
+            openedTag: null,
+            showTrash: false,
+            searchTags: tagsFromSearch(soFar),
+            titleOnly: true,
+          },
+          5
+        ).map(([noteId, note]) => ({
+          noteId,
+          content: note.content,
+          isPinned: note.systemTags.includes('pinned'),
+          ...noteTitleAndPreview(note),
+        }));
+
+        const additionalTextEdits =
+          precedingBracket >= 0
+            ? [
+                {
+                  text: '',
+                  range: {
+                    startLineNumber: position.lineNumber,
+                    startColumn: precedingBracket,
+                    endLineNumber: position.lineNumber,
+                    endColumn: position.column,
+                  },
+                },
+              ]
+            : [];
+
+        return {
+          incomplete: true,
+          suggestions: notes.map((note, index) => ({
+            additionalTextEdits,
+            kind: note.isPinned
+              ? languages.CompletionItemKind.Snippet
+              : languages.CompletionItemKind.File,
+            label: note.title,
+            // detail: note.preview,
+            documentation: note.content,
+            insertText: `[${note.title}](simplenote://note/${note.noteId})`,
+            sortText: index.toString(),
```

Comment:
I believe that this property is used to sort the results. sorting by index should then reflect the sort order in the notes list, as that's how we proceed through notes in the search.

if we want any kind of alphabetical sort we could use `note.title` and if we wanted some relevance search we could perform a quick search distance calculation on the note title against the search query. that would put closer matches at the top